microsoft/testsuites/network/common.py

```diff
@@ -222,3 +222,33 @@ def sriov_disable_enable(environment: Environment, times: int = 4) -> None:
         if sriov_is_enabled:
             sriov_basic_test(environment, vm_nics)
         network_interface_feature.switch_sriov(enable=(not sriov_is_enabled))
+
+
+def get_extra_nics(environment: Environment) -> Dict[str, int]:
+    nics_info: Dict[str, int] = {}
+    # get current nics status for restoring purpose
+    for node in environment.nodes.list():
+        network_interface_feature = node.features[NetworkInterface]
+        synthetic_nic_count = network_interface_feature.get_nic_count(
+            is_sriov_enabled=False, include_primary=False
+        )
+        sriov_nic_count = network_interface_feature.get_nic_count(
+            is_sriov_enabled=True, include_primary=False
+        )
+        nics_info[f"{node.name}_synthetic"] = synthetic_nic_count
+        nics_info[f"{node.name}_sriov"] = sriov_nic_count
+    return nics_info
+
+
+def restore_extra_nics(environment: Environment, nics_info: Dict[str, int]) -> None:
```

Comment:
The "remove" should be called in "restore".

Reply:
update